common/helpers/src/avro.rs

```diff
@@ -0,0 +1,248 @@
+		record_list.put(field_name, field_value.clone());
+	}
+	writer.append(record_list)?;
+	let serialized_bytes = writer.into_inner();
+	match serialized_bytes {
+		Ok(serialized_bytes) => Ok(serialized_bytes),
+		Err(error) => Err(AvroError::Avro(error)),
+	}
+}
+
+/// Function to populate a schema with serialized data and return an Avro reader.
+/// # Arguments
+/// * `schema` - Avro schema
+/// * `serialized_data` - serialized data
+/// # Examples
+/// ```
+/// use avro::*;
+/// let schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let record = vec![("name".to_string(), SchemaValue::String("John".to_string())), ("favorite_number".to_string(), SchemaValue::Int(42))];
+/// let schema_fingerprint = fingerprint_raw_schema(schema);
+/// assert!(schema_fingerprint.is_ok());
+/// let serialized_record = populate_schema_and_serialize(schema_fingerprint.unwrap().1, record);
+/// let reader = get_schema_data_reader(schema_fingerprint.unwrap().1, serialized_record.unwrap());
+/// ```
+pub fn get_schema_data_reader<'a>(
+	serialized_data: &'a Vec<u8>,
+	schema: &'a Schema,
+) -> Result<Reader<'a, &'a [u8]>, AvroError> {
+	Ok(Reader::with_schema(&schema, &serialized_data[..])?)
+}
+
+/// Function to read serialized data into a schema and return a hashmap of records(K,V).
+/// # Arguments
+/// * `schema` - Avro schema
+/// * `serialized_data` - serialized data
+/// # Returns
+/// * `Result<HashMap<String, SchemaValue>, AvroError>` - HashMap of records
+/// # Examples
+/// ```
+/// use avro::*;
+/// let schema = r#"{"type": "record", "name": "User", "fields": [{"name": "name", "type": "string"}, {"name": "favorite_number", "type": "int"}]}"#;
+/// let record = vec![("name".to_string(), SchemaValue::String("John".to_string())), ("favorite_number".to_string(), SchemaValue::Int(42))];
+/// let schema_fingerprint = fingerprint_raw_schema(schema);
+/// assert!(schema_fingerprint.is_ok());
+/// let serialized_record = populate_schema_and_serialize(schema_fingerprint.unwrap().1, record);
+/// let record_map = get_schema_data_map(schema_fingerprint.unwrap().1, serialized_record.unwrap());
+pub fn get_schema_data_map<'a>(
+    serialized_data: &'a Vec<u8>,
+    schema: &'a Schema,
+) -> Result<HashMap<String, SchemaValue>, AvroError> {
+    let reader = get_schema_data_reader(serialized_data, schema)?;
+    let mut result_record = HashMap::<String, SchemaValue>::new();
+    for record in reader {
+        let record_value = record.unwrap();
+        match record_value {
+            SchemaValue::Record(record) => {
+                for (field_name, field_value) in record.iter() {
+                    result_record.insert(field_name.clone(), field_value.clone());
+                }
+            }
+            _ => {
+                return Err(AvroError::InvalidRecords());
+            }
+        }
+    }
+    Ok(result_record)
+}
```

Comment:
return a hashmap of key value pair for a serialized data adhering to specific schema

Reply:
This is the only getter which accepts a serialized data of which was converted to avro datum and load it back in schema and return hashmap